test/examples/fortran_example.f90

```diff
@@ -0,0 +1,324 @@
+  end do
+
+  res = derivative(id, &
+                   num_points, &
+                   num_variables, &
+                   vector_length, &
+                   density)
+  deallocate(density)
+
+  ! compare with reference
+  call assert((abs(-9.4927931153398468d0 - res) < 1.0d-6), &
+              "derivatives do not match reference numbers")
+
+
+  !-----------------------------------------------------------------------------
+  ! now we try 3nd order, contracted with perturbed densities
+
+  order = 3
+  ierr = xc_eval_setup(id, XC_N_NX_NY_NZ, XC_CONTRACTED, order)
+  call assert(ierr == 0, "xc_eval_setup failed")
+
+  vector_length = 2**order
+  allocate(density(vector_length, num_variables, num_points))
+  density = 0.0d0
+  do ipoint = 1, num_points
+    ! we use fantasy values here
+    density(1, 1, ipoint) = 1.0d0   ! zeroth order
+    density(1, 2, ipoint) = 2.0d0   ! zeroth order
+    density(1, 3, ipoint) = 3.0d0   ! zeroth order
+    density(1, 4, ipoint) = 4.0d0   ! zeroth order
+    density(2, 1, ipoint) = 5.0d0   ! first order (1)
+    density(2, 2, ipoint) = 6.0d0   ! first order (1)
+    density(2, 3, ipoint) = 7.0d0   ! first order (1)
+    density(2, 4, ipoint) = 8.0d0   ! first order (1)
+    density(3, 1, ipoint) = 9.0d0   ! first order (2)
+    density(3, 2, ipoint) = 10.0d0  ! first order (2)
+    density(3, 3, ipoint) = 11.0d0  ! first order (2)
+    density(3, 4, ipoint) = 12.0d0  ! first order (2)
+    density(4, 1, ipoint) = 5.0d0   ! second order (depending on (1) and (2))
+    density(4, 2, ipoint) = 6.0d0   ! second order (depending on (1) and (2))
+    density(4, 3, ipoint) = 7.0d0   ! second order (depending on (1) and (2))
+    density(4, 4, ipoint) = 8.0d0   ! second order (depending on (1) and (2))
+    density(5, 1, ipoint) = 9.0d0   ! first order (3)
+    density(5, 2, ipoint) = 10.0d0  ! first order (3)
+    density(5, 3, ipoint) = 11.0d0  ! first order (3)
+    density(5, 4, ipoint) = 12.0d0  ! first order (3)
+    density(6, 1, ipoint) = 5.0d0   ! second order (depending on (1) and (3))
+    density(6, 2, ipoint) = 6.0d0   ! second order (depending on (1) and (3))
+    density(6, 3, ipoint) = 7.0d0   ! second order (depending on (1) and (3))
+    density(6, 4, ipoint) = 8.0d0   ! second order (depending on (1) and (3))
+    density(7, 1, ipoint) = 9.0d0   ! second order (depending on (2) and (3))
+    density(7, 2, ipoint) = 10.0d0  ! second order (depending on (2) and (3))
+    density(7, 3, ipoint) = 11.0d0  ! second order (depending on (2) and (3))
+    density(7, 4, ipoint) = 12.0d0  ! second order (depending on (2) and (3))
+    density(8, 1, ipoint) = 0.0d0   ! third order (depending on (1), (2) and (3))
+    density(8, 2, ipoint) = 0.0d0   ! third order (depending on (1), (2) and (3))
+    density(8, 3, ipoint) = 0.0d0   ! third order (depending on (1), (2) and (3))
+    density(8, 4, ipoint) = 0.0d0   ! third order (depending on (1), (2) and (3))
+  end do
+
```

Comment:
In the documentation it will be important to clarify
(1) the order of the densities and gradients in input
(2) the length of the output: I assume that also in the contracted case the output length might not be one single number, but rather all possible contractions up to the given order (correct me if I'm wrong).

Reply:
Can you please file an issue for this? These are very good questions but I would like the PR to go in before addressing them.

Reply:
Sure!